frontend/hub/useHubContext.tsx

```diff
@@ -0,0 +1,140 @@
+import { hubAPI } from './api';
+
+type HubFeatureFlags = {
+  // execution environments
+  container_signing: boolean;
+  execution_environments: boolean;
+
+  // keycloak login screen
+  external_authentication: boolean;
+
+  // community mode
+  ai_deny_index: boolean;
+  display_repositories: boolean;
+  legacy_roles: boolean;
+
+  // collection signing
+  can_create_signatures: boolean;
+  can_upload_signatures: boolean;
+  collection_auto_sign: boolean;
+  collection_signing: boolean;
+  display_signatures: boolean;
+  require_upload_signatures: boolean;
+  signatures_enabled: boolean;
+
+  // errors
+  _messages: string[];
+};
+
+type HubSettings = {
+  GALAXY_AUTO_SIGN_COLLECTIONS: boolean;
+  GALAXY_COLLECTION_SIGNING_SERVICE: string;
+  GALAXY_CONTAINER_SIGNING_SERVICE: string;
+  GALAXY_ENABLE_UNAUTHENTICATED_COLLECTION_ACCESS: boolean;
+  GALAXY_ENABLE_UNAUTHENTICATED_COLLECTION_DOWNLOAD: boolean;
+  GALAXY_MINIMUM_PASSWORD_LENGTH: number | null;
+  GALAXY_REQUIRE_CONTENT_APPROVAL: boolean;
+  GALAXY_REQUIRE_SIGNATURE_FOR_APPROVAL: boolean;
+  GALAXY_SIGNATURE_UPLOAD_ENABLED: boolean;
+  GALAXY_TOKEN_EXPIRATION: number | null;
+};
+
+type HubPermissions = {
+  [key: string]: {
+    global_description: string;
+    has_model_permission: boolean;
+    name: string;
+    object_description: string;
+    ui_category: string;
+  };
+};
+
+type HubGroup = {
+  id: number;
+  name: string;
+  object_roles?: string[];
+};
+
+type HubUser = {
+  auth_provider: string[];
+  date_joined: string;
+  email: string;
+  first_name: string;
+  groups: HubGroup[];
+  id: number;
+  is_anonymous: boolean;
+  is_superuser: boolean;
+  last_name: string;
+  model_permissions: HubPermissions;
+  username: string;
+};
+
+export type HubContext = {
+  errors: string[];
+  featureFlags: HubFeatureFlags;
+  hasPermission: (name: string) => boolean;
+  settings: HubSettings;
+  user: HubUser;
+};
+
+const HubContext = createContext<HubContext | null>(null);
+
+export const useHubContext = (): HubContext => useContext(HubContext) as HubContext;
+
+export const HubContextProvider = ({ children }: { children: ReactNode }) => {
+  const getFeatureFlags = useGet<HubFeatureFlags>(hubAPI`/_ui/v1/feature-flags/`);
+  const getSettings = useGet<HubSettings>(hubAPI`/_ui/v1/settings/`);
+  const getUser = useGet<HubUser>(hubAPI`/_ui/v1/me/`);
+
+  const [context, setContext] = useState<HubContext | null>(null);
+
+  useEffect(() => {
+    if (getFeatureFlags.isLoading || getSettings.isLoading || getUser.isLoading) {
+      return;
+    }
+
+    setContext({
+      errors: [
+        getFeatureFlags.error,
+        getSettings.error,
+        getUser.error,
+        ...(getFeatureFlags.data?._messages || []),
+      ].filter(Boolean) as string[],
+      featureFlags: getFeatureFlags.data as HubFeatureFlags,
+      settings: getSettings.data as HubSettings,
+      user: getUser.data as HubUser,
+    } as HubContext);
+  }, [getFeatureFlags, getSettings, getUser]);
```

Comment:
I tried testing by adding the line `const hubConfig = useHubContext();` in the Collections component and it was causing the page to get blocked due to too many re-renders.
![image](https://github.com/ansible/ansible-ui/assets/43621546/fdd1b57a-5e54-40a4-abcb-791400a4d2d3)

```suggestion
  }, [
    getFeatureFlags.isLoading,
    getSettings.isLoading,
    getUser.isLoading,
    getFeatureFlags.data,
    getSettings.data,
    getUser.data,
  ]);
```

Reply:
ah, I had it that way with `isLoading` but then the linter forced me to remove it or include all the .data and .error fields as well :)

Sounds like I have to do the latter, thanks :)